src/packs/walk_directory.rs

```diff
@@ -116,6 +251,19 @@ pub(crate) fn walk_directory(
                         let relative_path = child_absolute_dirname
                             .strip_prefix(cloned_absolute_root.as_ref())
                             .unwrap();
+
+                        // Check gitignore first (if enabled)
+                        if let Some(ref gitignore) = cloned_gitignore.as_ref() {
+                            let is_dir = child_dir_entry.file_type.is_dir();
```

Comment:
Given the check below:
```ruby
if gitignore.matched(&relative_path, false).is_ignore() {
    continue;
}
```
should we even attempt check `is_ignore` on line 260 if `is_dir == false`?